### PR TITLE
Add preliminary support for tables

### PR DESCRIPTION
**Pull request by pengjiz**
(Re: #2, #289)

Hello!

I added a very basic implementation for table rendering.

The implementation does not depend on any existing TTY table rendering libraries because I realized that we need to support, e.g., markups inside table cells and attributes, but I did not find a library that supports such a use case well.

Right now it is only a quick and dirty implementation. Many functionalities are missing, e.g., cell alignment, head row formatting, table rules, and markups inside cells. However, I noticed that you had created a `blockwise-rendering` branch, which might be very helpful for table rendering as well. So I am opening this PR to get some comments and hopefully provide some insights on implementing the new blockwise rendering.

Thanks!

PS here is the table in README rendered with this implementation:

mdcat-table

Nothing fancy indeed.

### CODE REVIEW

**Comment by swsnr** [comment]
Haven't taken a deeper look yet, but could you please not abbreviate table to tbl?  It's easier to read written out.

**Comment by pengjiz** [comment]
Done. I was doing so to avoid collision with pulldown-cmark tag 
enums, but indeed it looks much better now without the 
abbreviation.

File: pulldown-cmark-mdcat/src/render/data.rs
Comment (swsnr) [inline pulldown-cmark-mdcat/src/render/data.rs:71]: I'd appreciate if you'd place these impls right after the corresponding struct definition.

That's just how I prefer it in rust code 😇

File: pulldown-cmark-mdcat/src/render/write.rs
Comment (swsnr) [inline pulldown-cmark-mdcat/src/render/write.rs:367]: Unwrap always sets off my alarm; would you add a small comment above this line to explain why we can safely call unwrap here without risking a panic?

File: src/lib.rs
Comment (swsnr) [inline src/lib.rs:83]: Don't we also need to update all tests which parse markdown to enable tables?  So as to check that we don't break anything?

**Review by swsnr — CHANGES_REQUESTED** [review]
Thanks a lot for this patch 🙏❤️

I'd not like to hold this off with a lengthy review, since any support is better than no support 😅

I've just had a few minor remarks on the doff, and some overall points:

- I'd appreciate a small test, which renders a small file with two tables to assert the formatting and to check that we properly reset the current table state at the end of a table.  
- Please do add a changelog entry.
- Please check the manpage file and the README and the top level api docs whether we need to update them now that we support tables (I believe this restriction is mentioned in a couple of places). 


## Tests (edit)

Tests work mostly with golden files, i.e. the tests generate files which I check manually and then add to the repo; future test runs then compare against the existing golden file.

For this PR you'd add a e.g. `tables.md` to `pulldown-cmark-mdcat/tests/render/md/samples/`, then run the tests with `$MDCAT_UPDATE_GOLDEN_FILES` set to generate the initial golden file for `tables.md`.

Note that no other file should probably change, as tables aren't yet tested anywhere.  If any other golden file changes you'd need to investigate.

**Comment by pengjiz** [comment]
Thanks a lot for the review!

Honestly I did not expect that this PR would be merged – I was planning to rewrite it after your `blockwise-rendering` would get merged. If you'd like to install this implementation, I will address your comments and cleanup it a bit, perhaps over this weekend.

**Comment by swsnr** [comment]
If you wait for blockwise rendering you'll wait a long time. I don't have much time for this kind of stuff these days, so if I get this done this year, it's already fast 😅🙈

Let's merge what we have ✌️

**Comment by pengjiz** [comment]
Sorry for the delay. I got a bit busy in the past few weeks.

I just updated the implementation and then squashed and force pushed to my branch. It is ready from my side, and I would appreciate a review.

Thank you!

PS here is an updated screenshot rendering the README.md file:

mdcat-tables

**Comment by swsnr** [comment]
Thanks for finishing this. Now I'm a bit busy, but I hope to have a look by the end of this week.

If you haven't heard from me beginning of next week, please ping me 😊

**Review by swsnr — APPROVED** [review]

**Comment by swsnr** [comment]
@pengjiz  clippy complained 😬 Could take a look?  Once the pipeline is green, I'll merge and publish a new release

**Pull request merged**
